src/Core/MySQL/MySQLJsonConv.cpp

```diff
@@ -0,0 +1,940 @@
+    const auto key_length = getUInt16(data + entry_offset + offset_size);
+    return String(data + key_offset, key_length);
+}
+
+static void escape(String &in, String& out)
+{
+    size_t end_index = in.length();
+    for (size_t i = 0; i < end_index; ++i)
+    {
+        char c = in.at(i);
+        switch (c)
+        {
+        case '"':
+            out +=  "\\\"";
+            break;
+        case '\n':
+            out += "\\n";
+            break;
+        case '\r':
+            out += "\\r";
+            break;
+        case '\\':
+            out += "\\\\";
+            break;
+        case '\t':
+            out += "\\t";
+            break;
+        default:
+            out += c;
+            break;
+        }
+    }
+}
+
+
+static Int8 getInt8(const char * buffer)
+{
+    Int8 ret;
+    memcpy(&ret, buffer, sizeof(ret));
+    return ret;
+}
+
+static Int16 getInt16BE(const UInt8 * buf)
+{
+    return static_cast<Int16>(static_cast<UInt32>(buf[1]) +
+            (static_cast<UInt32>(buf[0]) << 8));
+}
+
+static Int32 getInt24BE(const UInt8 * buf)
+{
+    return static_cast<Int32>((buf[0] & 128) ? ((255U << 24) | (static_cast<UInt32>(buf[0]) << 16) |
+            (static_cast<UInt32>(buf[1]) << 8) | static_cast<UInt32>(buf[2]))
+            : ((static_cast<UInt32>(buf[0]) << 16) |
+            (static_cast<UInt32>(buf[1]) << 8) | (static_cast<UInt32>(buf[2]))));
+}
+
+static Int32 getInt32BE(const UInt8 * buf)
+{
+    return static_cast<Int32>(static_cast<UInt32>(buf[3]) +
+        (static_cast<UInt32>(buf[2]) << 8) +
+        (static_cast<UInt32>(buf[1]) << 16) +
+        (static_cast<UInt32>(buf[0]) << 24));
+}
+
+static void parseDecimalAndConv2String(const char * data, size_t data_len, String & out)
+{
+    /* decimal representation */
+    static constexpr Int32 DIG_PER_DEC  = 9;
+    static constexpr Int32 DIG_BASE      = 1000000000;
+    static constexpr Int32 DIG_MAX       = DIG_BASE - 1;
+    static constexpr Int32 dig2bytes[]   = { 0, 1, 1, 2, 2, 3, 3, 4, 4, 4 };
+    static constexpr Int32 powers10[]    = { 1, 10, 100, 1000, 10000, 100000, 1000000, 10000000, 100000000, 1000000000 };
+
+    static constexpr Int32 DIG_PER_INT32 = 9;
+    static constexpr Int32 SIZE_OF_INT32 = 4;
+
+    size_t cur_offset = 0;
+    Int32 precision = static_cast<UInt8>(getInt8(data));
+    cur_offset += 1;
+    Int32 scale = static_cast<UInt8>(getInt8(data + cur_offset));
+    cur_offset += 1;
+
+    Int32 intg = precision - scale;
+    Int32 frac = scale;
+    Int32 intg0 = intg / DIG_PER_INT32;
+    Int32 frac0 = frac / DIG_PER_INT32;
+    Int32 intg0x = intg - intg0 * DIG_PER_INT32;
+    Int32 frac0x = frac - frac0 * DIG_PER_INT32;
+
+    Int32 bin_size = intg0 * SIZE_OF_INT32 + dig2bytes[intg0x] + frac0 * SIZE_OF_INT32 + dig2bytes[frac0x];
+
+    if (bin_size + cur_offset > data_len)
+    {
+        throw DB::Exception(ErrorCodes::UNKNOWN_EXCEPTION, "illegal json data");
```

Comment:
The error message should be a well-formed sentence, starting with a capital letter, and it should contain the detailed explanation about the error.